ext/aws/freertos.md

```diff
@@ -0,0 +1,70 @@
+# FreeRTOS
+
+Amazon FreeRTOS is an open source, real-time operating system (RTOS) for
+microcontrollers in small, low-power devices.
+
+This module provides the latest FreeRTOS LTS release intergrated with modm:
+
+- Chooses the right Cortex-M port for the target.
+- Variable tick rate integrated with `modm::Clock` via `modm:platform:clock`.
+- Dynamic memory provided by the `modm:platform:heap` module with thread-safe
+  locking of Newlib's `malloc/free`.
+- Assertions integrated into the `modm:architecture:assert` interface.
+- Interrupt handling integrated with CMSIS NVIC functions.
+
+Note that we recommend using the FreeRTOS API directly and only if you write
+threaded code that also runs on other platforms (x86 simulators for example)
+should you consider using the API defined in `modm:processing:rtos` module.
+
+
+## Configuration
+
+This module generates a `FreeRTOSConfig.h` config file for modm integration with
+these settings:
+
+- `configASSERT(x)` implemented with `modm_assert(x, "freertos")`.
+- `configCPU_CLOCK_HZ` implemented with CMSIS `SystemCoreClock`.
+- `configTICK_RATE_HZ` set to `modm:freertos:frequency` or 1kHz on Cortex-M0.
+- `configSUPPORT_DYNAMIC_ALLOCATION = 1` as implemented by `modm:platform:heap`.
```

Comment:
Is there a reason why this cannot be a user setting?
(and the dependency on `modm:platform:heap` could be removed)

Reply:
It was like that before I didn't change it ;-P
You can force it by `#undef`ing the definition in the local config. However, I'm not sure how well FreeRTOS is usable without heap. At least the `modm:processing:rtos` code requires dynamic heap calls to `xSemaphoreCreateMutex` (vs. `xSemaphoreCreateMutexStatic`), `xQueueCreate`, `xTaskCreate` etc.